scene/2d/physics/collision_shape_2d.cpp

```diff
@@ -227,6 +229,18 @@ real_t CollisionShape2D::get_one_way_collision_margin() const {
 	return one_way_collision_margin;
 }
 
+void CollisionShape2D::set_one_way_collision_direction(Vector2 p_direction) {
+	one_way_collision_direction = p_direction.normalized();
```

Comment:
There should be a check whether the new direction is different from the old one, to avoid unnecessary callbacks and redraws.